### PR TITLE
Allow use of version qualifier for staging artifacts

### PR DESCRIPTION
**Pull request by dliappis**
## Proposed commit message

This commits adds support for an optional $VERSION_QUALIFIER env var/build option in the packaging pipeline to allow building prerelease artifacts from *staging*.

## How to test this PR locally

Tested a staging build using this PR with `VERSION_QUALIFIER="alpha1"` (also passed `DRA_BRANCH="main"` as this PR is based on main and we'd normally have tested alpha1 on main until 9.0 is cut): https://buildkite.com/elastic/beats-packaging-pipeline/builds/2138#01946b34-0e11-40b0-9610-cc647d2bea1e

Also tested a snapshot build (using `DRA_BRANCH="main"` and `RUN_SNAPSHOT="true"`) to ensure that an empty `--qualifier` option works as advertised by the release manager: https://buildkite.com/elastic/beats-packaging-pipeline/builds/2143

## Related issues

https://github.com/elastic/ingest-dev/issues/4857

### CODE REVIEW

File: .buildkite/packaging.pipeline.yml [hunk 1]
```diff
@@ -24,7 +24,9 @@ steps:
     key: start-gate-snapshot
 
   - name: Start of concurrency group for DRA Staging
-    if: build.branch =~ /^[0-9]+\.[0-9x]+\$/
+    # exceptionally allow building staging from main when VERSION_QUALIFIER is set, to allow prerelease testing
+    # TODO remove OR clause below and above comment, and only allow matching /^[0-9]+\.[0-9x]+\$/ for build.branch
+    if: build.branch =~ /^[0-9]+\.[0-9x]+\$/ || build.env('VERSION_QUALIFIER') != null
```
Comment (rowlandgeoff) [inline .buildkite/packaging.pipeline.yml:29]: Making sure I understand the comment... so you're planning to revert the regex changes once we're done with prerelease testing?

Comment (dliappis) [reply .buildkite/packaging.pipeline.yml:29]: Yes, as the intention is to use it only for releasable branches (in particular 9.0)

Comment (alexsapran) [reply .buildkite/packaging.pipeline.yml:29]: I know this will be triggered for every major release, and this does not happen often. But why would we want to revert it?

**Comment by dliappis** [comment]
After a bit of back n forth this seems to work fine (https://buildkite.com/elastic/beats-packaging-pipeline/builds/2138#01946b34-0e11-40b0-9610-cc647d2bea1e)
The missing piece was appending the `$VERSION_IDENTIFIER` to the `dependencies.csv` file as discussed in https://elastic.slack.com/archives/C01DJ3DC0AX/p1736952200875849

**Review by rowlandgeoff — APPROVED** [review]
Still LGTM. Thanks for testing it thoroughly.
